File: solutions/104_MaximumDepthofBinaryTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.dfs_max(root,0)
    
    def dfs_max(self, root, accum):
        if not root:
            return accum
        if not root.left and not root.right:
            return accum+1
        return max(self.dfs_max(root.left,accum+1), self.dfs_max(root.right, accum+1))

File: solutions/104_MaximumDepthofBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    function helper(head,acc){
        if(!head){
            return acc+0
        }
        if(!head.left && !head.right){
            return acc+1
        }
        return Math.max(helper(head.left, acc+1),helper(head.right,acc+1))
    }
    
    return helper(root,0)
};